src/js/media-theme-element.js

```diff
@@ -0,0 +1,99 @@
+import { Window as window } from './utils/server-safe-globals.js';
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import { TemplateInstance } from './utils/template-parts.js';
+import { processor } from './utils/template-processor.js';
+
+// Export Template parts for players.
+export * from './utils/template-parts.js';
+
+export class MediaThemeElement extends window.HTMLElement {
+  static observedAttributes = ['template'];
+  static processor = processor;
+
+  renderRoot;
+  renderer;
+  #template;
+
+  constructor() {
+    super();
+    this.renderRoot = this.attachShadow({ mode: 'open' });
+
+    const observer = new MutationObserver(() => this.render());
+    observer.observe(this, { attributes: true });
+  }
+
+  get mediaController() {
+    // Expose the media controller if API access is needed
+    return this.renderRoot.querySelector('media-controller');
+  }
+
+  get template() {
+    const templateId = this.getAttribute('template');
+    if (templateId) {
+      const template = this.getRootNode()?.getElementById(templateId);
+      if (template) return template;
+    }
+    return this.#template;
+  }
+
+  set template(element) {
+    this.#template = element;
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    if (attrName === 'template' && oldValue != newValue) {
+      if (this.template) {
+        // Transform short-hand if/partial templates to directive & expression.
+        this.template.content
+          .querySelectorAll('template[if],template[partial]')
+          .forEach((t) => {
+            let directive;
+
+            if (t.hasAttribute('if')) directive = 'if';
+
+            if (t.hasAttribute('partial')) directive = 'partial';
+
+            if (directive) {
+              t.setAttribute('directive', directive);
+              t.setAttribute('expression', t.getAttribute(directive));
+            }
+          });
+
+        this.createRenderer();
+      }
+    }
+  }
+
+  get props() {
```

Comment:
suggestion(non-blocking): Since we have an attributes mutation observer, we should be able to compute and store this as an optimization. Additionally, if we stored a `#props` object that we updated (similar to our `#state` in Mux Player), this is also where we could potentially support other "well-defined" props, a la passing in event handlers (or really any value, I suppose?)